cmd/kubeadm/app/util/tokens_test.go

```diff
@@ -22,30 +22,96 @@ import (
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
 )
 
-func TestTokenParseErrors(t *testing.T) {
+func TestTokenParse(t *testing.T) {
 	invalidTokens := []string{
 		"1234567890123456789012",
 		"12345.1234567890123456",
 		".1234567890123456",
 		"123456.1234567890.123456",
+		"Abcdef.1234567890123456",
 	}
 
 	for _, token := range invalidTokens {
 		if _, _, err := ParseToken(token); err == nil {
-			t.Errorf("generateTokenIfNeeded did not return an error for this invalid token: [%s]", token)
+			t.Errorf("ParseToken did not return an error for this invalid token: [%s]", token)
+		}
+	}
+
+	validTokens := []string{
+		"abcdef.1234567890123456",
+		"123456.AABBCCDDEEFFGGHH",
```

Comment:
The secret parcel should be OK since only the token ID is actually used for storing and looking up token secrets.